`TestRewriteJWT` flakiness

## Failure

#### Link(s) to logs

- https://github.com/gravitational/teleport.e/actions/runs/15923637591/job/44915872694

#### Relevant snippet

```
    testing.go:1267: TempDir RemoveAll cleanup: unlinkat /tmp/TestRewriteJWTtest_roles-and-traits_behavior2327395981/001/log/upload/streaming/default: directory not empty
--- FAIL: TestRewriteJWT/test_roles-and-traits_behavior (0.22s)
```

--- a/lib/srv/app/server_test.go
+++ b/lib/srv/app/server_test.go
@@ -184,12 +184,15 @@ func SetUpSuiteWithConfig(t *testing.T, config suiteConfig) *Suite {
 	require.NoError(t, err)
 	t.Cleanup(func() { s.authServer.Close() })
 
+	recordingMode := types.RecordOff
 	if config.ServerStreamer != nil {
-		_, err = s.authServer.AuthServer.UpsertSessionRecordingConfig(s.closeContext, &types.SessionRecordingConfigV2{
-			Spec: types.SessionRecordingConfigSpecV2{Mode: types.RecordAtNodeSync},
-		})
-		require.NoError(t, err)
+		recordingMode = types.RecordAtNodeSync
 	}
+
+	_, err = s.authServer.AuthServer.UpsertSessionRecordingConfig(s.closeContext, &types.SessionRecordingConfigV2{
+		Spec: types.SessionRecordingConfigSpecV2{Mode: recordingMode},
+	})
+	require.NoError(t, err)
 
 	s.tlsServer, err = s.authServer.NewTestTLSServer()
 	require.NoError(t, err)

